ors-api/src/main/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListener.java

```diff
@@ -102,22 +104,27 @@ record SourceFileElements(String repoBaseUrlString, String repoName, String loca
     }
 
     SourceFileElements extractSourceFileElements(String sourceFilePropertyValue) {
-        String repoBaseUrlString = null;
-        String repoName = null;
-        String localOsmFilePath = "";
+        String accessScheme = null, repoBaseUrlString = null, repoName = null;
+        sourceFilePropertyValue = sourceFilePropertyValue.trim();
         try {
-            new URL(sourceFilePropertyValue);
-            LOGGER.debug("Configuration property 'source_file' contains a URL, using value as URL for a graphs repository");
-            sourceFilePropertyValue = sourceFilePropertyValue.trim().replaceAll("/$", "");
-            String[] urlElements = sourceFilePropertyValue.split("/");
+            URI uri = new URI(sourceFilePropertyValue);
+            accessScheme = uri.getScheme();
+            repoBaseUrlString = uri.getHost();
+            if (accessScheme == null || repoBaseUrlString == null) {
+                throw new URISyntaxException(sourceFilePropertyValue, "URI does not contain a valid schema or host");
+            }
+            repoBaseUrlString = accessScheme + "://" + repoBaseUrlString;
```

Comment:
this won't work, you cut off too much of the configured url - see my comment for the test for this method